torch/CMakeLists.txt

```diff
@@ -265,73 +183,145 @@ if (NOT NO_API)
     ${TORCH_SRC_DIR}/csrc/api/src/optim/lbfgs.cpp
     ${TORCH_SRC_DIR}/csrc/api/src/optim/rmsprop.cpp
     ${TORCH_SRC_DIR}/csrc/api/src/optim/sgd.cpp
-  )
+    )
+
 endif()
 
 add_library(torch SHARED ${TORCH_SRCS})
 
-# https://gcc.gnu.org/onlinedocs/gcc-4.0.3/gcc/Warning-Options.html
-target_compile_options(torch
-  PRIVATE
-  -Wall
-  -Wextra
-  -pedantic
-  -Wcast-align
-  -Wcast-qual
-  -Wctor-dtor-privacy
-  -Wdisabled-optimization
-  -Winit-self
-  -Wmissing-include-dirs
-  -Woverloaded-virtual
-  -Wsign-promo
-  -Wstrict-overflow=5
-  -Wundef
-  -fdiagnostics-show-option
-  -Wno-unused-parameter
-  -Wno-missing-braces # This warning is buggy
-  -Wno-unknown-pragmas)
-
-if ($ENV{WERROR})
+# until they can be unified, keep these lists synced with setup.py
+if(MSVC)
+  target_compile_options(torch PRIVATE
+    /MD
+    /Z7
+    /EHa
+    /DNOMINMAX
+    /wd4267
+    /wd4251
+    /wd4522
+    /wd4522
+    /wd4838
+    /wd4305
+    /wd4244
+    /wd4190
+    /wd4101
+    /wd4996
+    /wd4275
+    /bigobj
+    )
+else()
+  target_compile_options(torch PRIVATE
+    -std=c++11
+    -Wall
+    -Wextra
+    -Wno-unused-parameter
+    -Wno-missing-field-initializers
+    -Wno-write-strings
+#    -Wno-zero-length-array
```

Comment:
What's with `Wno-zero-length-array`?

Reply:
It breaks one of the build configurations, I think maybe it was that clang doesn't acknowledge that this flag exists?

Reply:
It shouldn't be commented out either way I think, let's remove it if we don't need it

Reply:
++

Reply:
kicked off this test run to check exactly what it breaks https://github.com/pytorch/pytorch/pull/9427

Reply:
error is at https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-linux-trusty-py3.6-gcc7-build/1833/console